lang/src/org/partiql/lang/util/NumberExtensions.kt

```diff
@@ -111,7 +111,8 @@ fun Number.ionValue(ion: IonSystem): IonValue = when (this) {
 
 operator fun Number.unaryMinus(): Number {
     return when (this) {
-        is Long -> -this
+        // - LONG.MIN_VALUE will result in LONG.MIN_VALUE in JVM because LONG is a signed two's-complement integers
+        is Long -> if (this == Long.MIN_VALUE) BigInteger.valueOf(Long.MAX_VALUE).add(BigInteger.ONE) else -this
```

Comment:
Please be kind to educate me here! 
 
For negating the Long.MIN_VALUE, this returns an integer that is 1 more than is representable for Long, so the result is of type BigInteger, not Long as was the input.  (Smells fishy, but ok). 

But that's a number as the result.  How does then the newly-uncommented test on Line 222 manage to succeed?  It seems to expect MISSING, not a proper number!

Reply:
@vgapeyev hope I've understood your question correctly: since this is being tested under _permissive_ mode upon receiving [the exception](https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L175) `Thunk` returns the _MISSING_ value, see [here](https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/src/org/partiql/lang/eval/Thunk.kt#L708). For the other point regarding BigInt unfortunately, I haven't been able to review this PR so can't really comment on that at the moment.

Reply:
Also, exception, if my understanding is correct, happens during "evaluation time". 

https://github.com/partiql/partiql-lang-kotlin/blob/0b4540f474caff7dba1d6c327e0e85012402829b/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L550-L575

Reply:
Yep, the permissive mode would explain this, thanks!